### PR TITLE
RHCLOUD-44391: migrate additional navigation tests from IQE

### PR DESCRIPTION
**Pull request by catastrophe-brandon**
## Dependencies
⚠️ **This PR depends on #3498 (btweed/fix-settings-menu-test)** - Must be merged first.

This branch has been rebased on top of `btweed/fix-settings-menu-test` to include:
- Isolated auth context cleanup fixes
- AUTH_TIMEOUT fixture improvements for CI environments

---

## Summary
**Supplemental to PR #3496**: Completes the migration of `test_navigation.py` from IQE by adding platform link validation tests and documenting chrome vs tenant testing boundaries.

### Relationship to Other PRs
This PR builds on **PR #3496** which migrated:
- Navigation toggle hide/show tests
- Navigation selection persistence after refresh
- Tekton pipeline fixes

This PR adds the remaining applicable tests from `test_navigation.py` and documents why some tests are NOT migrated.

---

#### Platform Service Links (3 test cases)
- **test_services_menu_platform_links**: Validates chrome's platform link navigation structure
  - Platform link - Ansible has correct internal route (`/ansible`)
  - Platform link - OpenShift has correct internal route (`/openshift/overview`)
  - Platform link - Insights has correct internal route (`/insights`)

**Testing Approach:** Validates link existence and `href` attributes using OUIA component IDs, without clicking through to destinations (which may not be available in CI). This tests that **chrome's navigation structure is correctly configured**, not whether destination routes are registered or functional.

#### Fancy 404 Page (1 test case)
- **test_404s**: Tests fancy 404 error page and "Return to homepage" button functionality

---

### Tests NOT Migrated - Chrome vs Tenant Boundaries

The following tests from `test_navigation.py` are **intentionally not migrated** because they test tenant application functionality, not chrome navigation:

#### ❌ test_services_menu_destinations
**Why not migrated:** Tests navigation to dynamically-generated service destinations (Cost Management, Vulnerability, Drift, etc.) which are **tenant applications**. Each tenant team is responsible for testing their own app's navigation and functionality.

**Chrome's responsibility:** Provide the services menu framework  
**Tenant's responsibility:** Verify their app is accessible and functional

#### ❌ test_non_services_menu_destinations
**Why not migrated:** Similar to above - tests navigation to tenant application routes outside the services menu. Tenant teams should verify their own applications.

#### ❌ test_broken_links
**Why not migrated:** Crawls all links on pages and verifies HTTP 200 responses. This is:
1. **Too broad** - covers tenant application content
2. **Too slow** - makes HTTP requests for every link
3. **Prone to false positives** - external links, auth boundaries, etc.

**Tenant responsibility:** Each app should test their own links and content

---

### Chrome vs Tenant Testing Philosophy

**Chrome tests:**
- Chrome navigation structure (menus, toggles, navigation framework)
- Chrome UI components (topbar, sidebar, breadcrumbs)
- Link configuration (`href` values, accessibility)

**Tenants test:**
- Their app's navigation works
- Their routes resolve correctly
- Their content and links are functional

**This separation:**
- Keeps chrome tests focused, fast, and maintainable
- Prevents duplication (chrome testing things tenants should test)
- Clarifies ownership and responsibility

---

### Changes
- Added 4 new test cases to `navigation.spec.ts`
- Updated `MIGRATION-SUMMARY.md` with:
  - Complete test_navigation.py migration analysis
  - Testing approach for platform links (structure validation vs destination testing)
  - Clear documentation of tests NOT migrated and why
  - Chrome vs tenant testing responsibility boundaries

### Migration Statistics
- **Tests Migrated:** 2 test functions → 4 test cases
- **Tests Skipped:** 3 (tenant responsibilities)

### IQE Source
Tests analyzed from: `iqe-platform-ui-plugin/iqe_platform_ui/tests/test_navigation.py`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Tests**
  * Added E2E tests validating Services menu links for Ansible, OpenShift, and Insights.
  * Added E2E test verifying the error page "Return to homepage" recovery flow.

* **Documentation**
  * Added a migration summary documenting migrated vs skipped tests and scoping decisions for tenant/destination checks.

* **Chores**
  * Simplified dev-server overlay behavior to suppress build overlay messages.

### CODE REVIEW

File: playwright/e2e/MIGRATION-SUMMARY.md [hunk 1]
```diff
@@ -1,5 +1,60 @@
...
+**Playwright Implementation:** 1 test case
+- ✅ Fancy 404 page returns to homepage
+
+Tests that the fancy 404 error page displays correctly and the "Return to homepage" button works.
+
+### Tests NOT Migrated (Chrome-Specific Reasoning)
+
+#### ❌ test_services_menu_destinations
+**Reason:** Tenant application responsibility
+**Details:** This test uses dynamic test generation to navigate through all service destinations registered in the platform. These destinations represent individual tenant applications (e.g., Cost Management, Drift, Vulnerability, etc.). Testing navigation within tenant apps is the responsibility of those respective teams. Chrome should only test that the chrome navigation framework works (which is covered by the platform links tests).
```
Comment (catastrophe-brandon) [inline playwright/e2e/MIGRATION-SUMMARY.md:33]: Note: We were previously doing this for tenant teams, but I think it's time that we expect tenant teams to conduct their own validation of their services menu contributions; we can't really know what the "correct" data is without a lot of contact chasing.

File: playwright/e2e/release-gate/settings.spec.ts [hunk 2]
```diff
@@ -47,52 +47,31 @@ test.describe('Settings Gear and Navigation', () => {
...
     ];
 
-    for (const expectedItem of expectedItems) {
-      // Check if any menu item contains the expected text (handles badges/extra text)
-      const found = menuItems.some(item => item.includes(expectedItem));
-      expect(found, `Expected to find "${expectedItem}" in menu items: ${JSON.stringify(menuItems)}`).toBe(true);
+    for (const ouiaId of expectedOuiaIds) {
+      const isVisible = await topbar.hasSettingsMenuItem(ouiaId);
+      expect(isVisible, `Expected settings menu item with OUIA ID "${ouiaId}" to be visible`).toBe(true);
     }
```
[the remark this answers is not in the document]
Comment (catastrophe-brandon) [reply playwright/e2e/release-gate/settings.spec.ts:64]: We don't run these tests against the ITLess environment, so this is not an issue in PR checks.

File: playwright/e2e/release-gate/navigation.spec.ts [hunk 3]
```diff
@@ -72,4 +72,69 @@ test.describe('Navigation', () => {
...
+    await page.locator('.chr-c-link-service-toggle').click();
+    await expect(page.locator('.pf-v6-c-sidebar__content')).toBeVisible();
+
+    // Find Ansible platform link within services menu using OUIA ID
+    const ansibleLink = page.locator('[data-ouia-component-id="AllServices-Dropdown-Ansible"]');
+
+    // Verify link exists and has correct internal href
+    await expect(ansibleLink).toBeVisible();
+    await expect(ansibleLink).toHaveAttribute('href', /\/ansible$/);
+  });
```
[the remark this answers is not in the document]
Comment (catastrophe-brandon) [reply playwright/e2e/release-gate/navigation.spec.ts:90]: We don't run these tests in the ITLess environment.

**Comment by catastrophe-brandon** [comment]
@coderabbit-ai resolve

**Review by justinorringer — APPROVED** [review]
LGTM!

**Review by karelhala — APPROVED** [review]
